### PR TITLE
Wait group for posts processing

### LINKED ISSUE #7: Concurrency problem
```
        for _, file := range files {
		if path.Ext(file.Name()) == ".md" || path.Ext(file.Name()) == ".markdown" {
			fmt.Println("Processing " + file.Name())
			post := createPost(file.Name())
			go createPostFile(post)
			posts = append(posts, post)
			fmt.Println("Done with  " + file.Name())
			fmt.Println("---")
		}
	}
```
Here you put `post` into `posts` just after runing the `createPostFile` goroutine, so you can't be sure that processing is already done, when you will try to make

```
t, _ := template.ParseFiles("index.html")
	t.Execute(f, posts)
```

In this case, you should use `channels` and `wait group` to make this code thread-safe

### PR DESCRIPTION
**Pull request by egregors**
resolve #7

### CODE REVIEW

**Comment by egregors** [comment]
@freetonik this is a really big bunch of updates, but I believe these changes will make the project much more extendable and reliable.

So,
- the project structure is idiomatically for golang now
- a small pinch of "OOP" (Post, Blog structs and methods) **nobody like "OOP", I know**
- all Post are processing in separate goroutines through the channel (CSP)
- workers limit defines according to Rlimit / 2 if it possible (macOS, Linux), otherwise (Win) just use default limit (100)
- thread-safe append posts

File: app/cmd/post.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	month := filenameBase[3:5]
+	year := filenameBase[6:10]
+	date, err := time.Parse("02-01-2006", day+"-"+month+"-"+year)
+	if err != nil {
+		log.Fatal(err)
+	}
+	slug := filenameBase[11:]
+
+	// Get body from file
+	mdfile, err := os.Open("./markdown/" + filename)
```
Comment (freetonik) [inline app/cmd/post.go:33]: I guess this func should respect `DefaultMarkdownPath`, too. Accept path as param?

**Comment by freetonik** [comment]
Whoa, amazing!

File: app/cmd/post.go (hunk 1 above)
Comment (egregors) [reply app/cmd/post.go:33]: @freetonik  Yep, you are right. This place should be refactored during the #10 implementation

**Pull request merged**